geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -332,21 +331,22 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
         case Seq(f) => Some(f)
         case f      => Some(new FilterList(f: _*))
       }
-      ranges.map { r =>
-        val scan = new Scan(r.getStartRow, r.getStopRow)
-        scan.addFamily(colFamily).setCacheBlocks(cacheBlocks).setSmall(true)
-        filter.foreach(scan.setFilter)
-        cacheSize.foreach(scan.setCaching)
-        ds.applySecurity(scan)
-        scan
+      // note: we have to copy the ranges for each table scan
+      tables.map { table =>
+        val scans = ranges.map { r =>
+          val scan = new Scan(r.getStartRow, r.getStopRow)
+          scan.addFamily(colFamily).setCacheBlocks(cacheBlocks).setSmall(true)
+          filter.foreach(scan.setFilter)
+          cacheSize.foreach(scan.setCaching)
+          ds.applySecurity(scan)
+          scan
+        }
+        TableScan(table, scans)
```

Comment:
It looks like this bit of separation *could* allow us to do something clever later, right?

Reply:
what did you have in mind? if we could tell whether there was actually data there for a given table, maybe we could trim down the ranges for that table, but i don't think there is any way to do that.

Reply:
Well, just that if we did have a function that'd do what we wanted, we could use it above, etc, etc.